ts/lib220.ts

```diff
@@ -1,5 +1,79 @@
 import { getRunner } from './runtime';
 
+function hexColorChannel(n: number): string {
+  let v = (Math.floor(n * 255)).toString(16);
+  if (v.length < 2) v = '0' + v;
+  return v;
+}
+
+function rgbToHex(rgb: number[]): string {
+  let hex = '#';
+  for (let i = 0; i < 3; ++i) {
+    hex += hexColorChannel(rgb[i]);
+  }
+  return hex;
+}
+
+export class DrawingCanvas {
+  width: number = 1;
+  height: number = 1;
+  ctx: CanvasRenderingContext2D | undefined = undefined;
```

Comment:
Just write ` ctx: CanvasRenderingContext2D;`

TypeScript will check the constructor and see that ctx is initialized (strict class initialization):

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html

Reply:
That won't work since `ctx` does not get set if this function is called under node.

Reply:
OK, this is fine. We should just merge. But, if we want students to actually use this, the `newCanvas` function should return a mock object that silently fails.

Reply:
Oh, true. Done.